### PR TITLE
 Update contributing docs

### LINKED ISSUE #1572: Add better contribute docs
Improve our contribute docs, e.g add information about outreachy and get inspired by our wonderful web-ext docs.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This got heavily inspired by the web-ext contribution docs, kudos for
them!

Fix #1572 

* remove grunt commands that are already documented via 'scripts' section

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,20 +1,144 @@
-Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
+Hi! Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
```
Comment (tofumatt) [inline CONTRIBUTING.md]: tenor

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,20 +1,144 @@
...
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+# Picking an issue
+
+If you're looking for a small task to work on so you can get familiar with the
+process of contributing patches, have a read through these
+[good first bugs](https://github.com/mozilla/web-ext/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+bug%22).
+
+If you are already familiar with the process or want to contribute even more we appreciate any help on our [contrib: welcome](https://github.com/mozilla/addons-linter/issues?q=is%3Aissue+is%3Aopen+label%3A"contrib%3A+welcome) issues.
```
Comment (muffinresearch) [inline CONTRIBUTING.md]: Maybe something like this:

If you're already familiar with the project or would like take on something a little more challenging, please take a look at the [contrib: welcome](https://github.com/mozilla/addons-linter/issues?q=is%3Aissue+is%3Aopen+label%3A"contrib%3A+welcome) issues.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,20 +1,144 @@
-Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
+Hi! Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
 
-## Submitting a Pull Request
+The linter is used in various stages of [WebExtension](https://developer.mozilla.org/en-US/Add-ons/WebExtensions)
```
Comment (tofumatt) [inline CONTRIBUTING.md]: This sentence is phrased a bit awkwardly and it makes it hard to parse. Also I'm pretty sure we're not supposed to say WebExtensions? I dunno. Saying "contributions are awesome" is a bit of a generic call-to-action as well, but I'd encourage people involved with extensions to relay their experience and get involved... I'd try:

```
This linter is used to help develop and publish [extensions](https://developer.mozilla.org/en-US/Add-ons/WebExtensions) for Firefox. You're an add-on developer, we would really value your contributions–no one knows add-on development and publishing better than an add-on developer!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,20 +1,144 @@
...
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
-Our tests include `eslint` style checks; these keep our code consistent.
-You can review the rules in the [`.eslintrc`][eslint] file.
 
-Please run the tests locally with `npm test` before you commit.
+- [Picking an issue](#picking-an-issue)
+- [Installation](#installation)
+  - [Develop all the things](#develop-all-the-things)
```
Comment (tofumatt) [inline CONTRIBUTING.md]: I know that "All the things" is a cute meme but not everyone gets memes, especially non-Western, English-speaking folks. This could be better written as "Development environment".

But actually looking at that section I think it can be removed and folded into testing.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,20 +1,144 @@
...
+  - [Run all application tests](#run-all-application-tests)
```
Comment (tofumatt) [inline CONTRIBUTING.md]: This should be "Testing the linter" and the "develop all the things" section should be moved under it to "Run tests when you make changes"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,20 +1,144 @@
...
+  - [Writing commit messages](#writing-commit-messages)
+  - [Tips](#tips)
 
-[eslint]: https://github.com/mozilla/addons-linter/blob/master/.eslintrc
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+# Picking an issue
+
+If you're looking for a small task to work on so you can get familiar with the
```
Comment (tofumatt) [inline CONTRIBUTING.md]: I'd change this whole paragraph to:

```md
For first-time contributors or those who want to start with a small task: [check out our list of good first bugs](https://github.com/mozilla/web-ext/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+bug%22). These issues have an assigned mentor to help you out and are great issues for learning about the linter and our development process.
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (tofumatt) [inline CONTRIBUTING.md]: "with the process" -> "with our development process"

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,20 +1,144 @@
...
+If you'd like to work on a bug, please comment on it to let the maintainers know.
+If someone else has already commented and taken up that bug, please refrain from working on it and submitting
+a PR without asking the maintainers as it leads to unnecessary duplication of effort.
+
+
+# Installation
+
+To get started on a patch, first install `addons-linter` from [source](README.md#development).
+
+## Develop all the things
```
Comment (tofumatt) [inline CONTRIBUTING.md]: Again, this should be under testing and could simply be named "Run tests when you make changes"

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,20 +1,144 @@
...
+
+Your one stop command to continuously run tests and check for
```
Comment (tofumatt) [inline CONTRIBUTING.md]: This is a bit too colloquial without being illuminating 😉 Maybe just:

```md
To run tests and check for JavaScript syntax issues as you change the code, run:

    npm test
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,20 +1,144 @@
...
+JavaScript syntax problems.
+
+    npm test
+
+The other commands below are just variations on this.
```
Comment (tofumatt) [inline CONTRIBUTING.md]: I don't think you need this line.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,20 +1,144 @@
...
+[pull request](https://help.github.com/articles/creating-a-pull-request/)
+for a new fix or feature, be sure to mention the issue
+number for what you're working on.
+The best way to do it is to mention the issue like
+this at the top of your description:
+
+    Fixes #123
+
+The issue number in this case is "123."
+The word *Fixes* is magical; github will automatically close the issue when your
```
Comment (tofumatt) [inline CONTRIBUTING.md]: "github" -> "GitHub"

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,20 +1,144 @@
...
+pull request is merged.
+
+## Writing commit messages
+
+Good commit messages serve at least three important purposes:
+
+* To speed up the reviewing process.
```
Comment (tofumatt) [inline CONTRIBUTING.md]: The "to" in front of these bullet points aren't quite right... I'd say "they" instead of "to".

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,20 +1,144 @@
...
+* To help us write a good release note.
+* To help the future maintainers of addons-linter (it could be you!), say five years into the future, to find out why a particular change was made to the code or why a specific feature was added.
```
Comment (tofumatt) [inline CONTRIBUTING.md]: This sentence is too long especially compared to the others. I'd just say "They help future maintainers understand your change and the reasons behind it."

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -1,20 +1,144 @@
...
+The issue number in this case is "123."
+The word *Fixes* is magical; github will automatically close the issue when your
+pull request is merged.
+
+## Writing commit messages
+
+Good commit messages serve at least three important purposes:
+
+* To speed up the reviewing process.
+* To help us write a good release note.
```
Comment (tofumatt) [inline CONTRIBUTING.md]: "They help us write good release notes"

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -1,20 +1,138 @@
-Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
+Hi! Thanks for wanting to contribute to Mozilla's Add-ons Linter! You rock! 😊
 
-## Submitting a Pull Request
+This linter is used to help develop and publish [Add-ons](https://developer.mozilla.org/en-US/Add-ons/) for Firefox. You're an add-on developer, we would really value your contributions–no one knows add-on development and publishing better than an add-on developer!
```
Comment (pdehaan) [inline CONTRIBUTING.md]: You may want to remove the `/en-US/` directory and let MDN redirect to the user's preferred locale.

Comment (EnTeQuAk) [reply CONTRIBUTING.md]: oh good point, damn copy & paste. Thanks!
